integration/core/test_cli.py

```diff
@@ -798,11 +798,63 @@ def backup_core(bin, engine_manager_client,  # NOQA
     assert backup_list[backup2]["Size"] == backup2_info["Size"]
     assert backup_list[backup2]["Created"] == backup2_info["Created"]
 
-    # test backup volume list
-    # https://github.com/rancher/longhorn/issues/399
+    # test that corrupt backups don't show up in the list
+    # https://github.com/longhorn/longhorn/issues/1212
+    cmd = [bin, '--url', grpc_controller_client.address,
+           'backup', 'ls', backup_target]
+    data = subprocess.check_output(cmd, env=env)
+    volume_info = json.loads(data)
+
+    assert volume_info
+    assert volume_info[VOLUME_NAME] is not None
+    assert volume_info[VOLUME_NAME]["Backups"] is not None
+    backups = volume_info[VOLUME_NAME]["Backups"]
+    assert len(backups) == 2
+    assert backup1_info["URL"] in backups
+    assert backup2_info["URL"] in backups
+
     volume_dir = finddir(BACKUP_DEST, VOLUME_NAME)
     assert volume_dir
     assert path.exists(volume_dir)
+    backup_dir = os.path.join(volume_dir, "backups")
+    assert path.exists(backup_dir)
+    backup_cfg_name = "backup_" + backup2_info["Name"] + ".cfg"
+    assert backup_cfg_name
+    backup_cfg_path = findfile(backup_dir, backup_cfg_name)
+    assert path.exists(backup_cfg_path)
+    backup_tmp_cfg_path = backup_cfg_path.replace(".cfg", ".bak")
+    os.rename(backup_cfg_path, backup_tmp_cfg_path)
+    assert path.exists(backup_tmp_cfg_path)
+
+    corrupt_backup = open(backup_cfg_path, "w")
+    assert corrupt_backup
+    assert corrupt_backup.write("{corrupt: definitely") > 0
+    corrupt_backup.close()
+
+    # request the new backup list
+    data = subprocess.check_output(cmd, env=env)
+    volume_info = json.loads(data)
+    assert volume_info
+    assert volume_info[VOLUME_NAME] is not None
+    assert volume_info[VOLUME_NAME]["Backups"] is not None
+    backups = volume_info[VOLUME_NAME]["Backups"]
+    assert len(backups) == 1
+    assert backup1_info["URL"] in backups
+    assert backup2_info["URL"] not in backups
+
+    # will fail because of corrupt json data
+    cmd = [bin, '--url', grpc_controller_client.address,
+           'backup', 'inspect', backup2_info["URL"]]
+    with pytest.raises(subprocess.CalledProcessError):
```

Comment:
Shoud `inspect` fail or return some error message in the json?

Reply:
By doing with with pytest.raises, it will check that the command has failed.
Which happens on the engine side when an error is returned.

Reply:
Since inspect is trying to load a backup.cfg that either doesn't exist or something went wrong during loading. Both cases lead to failure of inspect.

Both cases after the fix will be ignored, since we don't want the full list to fail just because a single backup.cfg couldn't be loaded.